youtube_dl/extractor/rai.py

```diff
@@ -55,7 +55,118 @@ def _extract_relinker_formats(self, relinker_url, video_id):
 
         return formats
 
-    def _extract_from_content_id(self, content_id, base_url):
+class RaiPlayIE(RaiBaseIE):
+    _VALID_URL = r'https?://(?:www\.)?raiplay\.it/.+?-(?P<id>[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12})\.html'
+    _TESTS = [{
+        'url': 'http://www.raiplay.it/video/2016/10/La-Casa-Bianca-e06118bb-59a9-4636-b914-498e4cfd2c66.html',
+        'md5': '340aa3b7afb54bfd14a8c11786450d76',
+        'info_dict': {
+            'id': 'e06118bb-59a9-4636-b914-498e4cfd2c66',
+            'ext': 'mp4',
+            'title': 'La Casa Bianca',
+            'thumbnail': 're:^https?://.*\.jpg$',
+            'uploader': 're:^Rai.+',
+            'description': 're:^[A-Za-z]+'
+        }
+    }, {
+        'url': 'http://www.raiplay.it/video/2016/11/gazebotraindesi-efebe701-969c-4593-92f3-285f0d1ce750.html',
+        'md5': 'ed4da3d70ccf8129a33ab16b34d20ab8',
+        'info_dict': {
+            'id': 'efebe701-969c-4593-92f3-285f0d1ce750',
+            'ext': 'mp4',
+            'title': '#gazebotraindesi',
+            'thumbnail': 're:^https?://.*\.png$',
+            'uploader': 're:^Rai.+',
+            'description': 're:^[A-Za-z]+'
+        }
+    }, {
+        'url': 'http://www.raiplay.it/video/2014/04/Report-del-07042014-cb27157f-9dd0-4aee-b788-b1f67643a391.html',
+        'md5': '8970abf8caf8aef4696e7b1f2adfc696',
+        'info_dict': {
+            'id': 'cb27157f-9dd0-4aee-b788-b1f67643a391',
+            'ext': 'mp4',
+            'title': 'Report del 07/04/2014',
+            'thumbnail': 're:^https?://.*\.jpg$',
+            'uploader': 're:^Rai.+',
+            'description': 're:^[A-Za-z]+'
+        }
+    }]
+    _RESOLUTION = '600x400'
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+
+        canonical_url = self._og_search_url(webpage)
```

Comment:
What's the point of this? `og:url` is always the same as the original `url`. If not provide test for that.